orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Stage.java

```diff
@@ -422,6 +422,44 @@ public Task taskById(String taskId) {
     }
   }
 
+  /**
+   * Get all stages that are descendants of the current one
+   */
+  public List<Stage> descendants() {
+    if (execution != null) {
+      HashSet<String> visited = new HashSet<>();
+      LinkedList<Stage> queue = new LinkedList<>();
+      List<Stage> descendents = new ArrayList<>();
+
+      queue.push(this);
+      boolean first = true;
+
+      while (!queue.isEmpty()) {
+        Stage stage = queue.pop();
+        if (!first) {
+          descendents.add(stage);
+        }
+
+        first = false;
+        visited.add(stage.refId);
+
+        List<Stage> childStages = stage.downstreamStages();
+
+        childStages
+          .stream()
+          .filter(s -> !visited.contains(s.refId))
+          .forEach(s -> queue.add(s));
+      }
+
+      return ImmutableList
```

Comment:
Might suggest not returning an Immutable or Unmodifiable list here ... the existing `downstreamStages()` doesn't and it's probably worth being consistent. 

We've been burnt one too many times with something being unmodifiable and blowing up at runtime when someone unexpectedly tries to add.

Reply:
I was following the pattern from `ancestors()` but don't have a preference, can also chance return type to `ImmutableList<>`? what would you prefer?